modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_component.dm

```diff
@@ -0,0 +1,273 @@
+///Global list containing any and all soulcatchers
+GLOBAL_LIST_EMPTY(soulcatchers)
+
+#define SOULCATCHER_EMOTE_COLOR "#4ba1c9"
+#define SOULCATCHER_SAY_COLOR "#75D5E1"
+#define SOULCATCHER_WARNING_MESSAGE "You have entered a soulcatcher, do not share any information you have received while a ghost. If you have died within the round, you do not know your identity until your body has been scanned, standard blackout policy also applies."
+
+/datum/component/soulcatcher
+	/// What is the name of the soulcatcher?
+	var/name = "soulcatcher"
+	/// What rooms are linked to this soulcatcher
+	var/list/soulcatcher_rooms = list()
+	/// Are ghosts currently able to join this soulcatcher?
+	var/ghost_joinable = TRUE
+
+/datum/component/soulcatcher/New()
+	. = ..()
+	if(!parent)
+		return COMPONENT_INCOMPATIBLE
+
+	create_room()
+	GLOB.soulcatchers += src
+
+/datum/component/soulcatcher/Destroy(force, ...)
+	GLOB.soulcatchers -= src
+	for(var/datum/soulcatcher_room in soulcatcher_rooms)
+		soulcatcher_rooms -= soulcatcher_room
+		qdel(soulcatcher_room)
+
+	return ..()
+
+/**
+ * Creates a `/datum/soulcatcher_room` and adds it to the `soulcatcher_rooms` list.
+ *
+ * Arguments
+ * * target_name - The name that we want to assign to the created room.
+ * * target_desc - The description that we want to assign to the created room.
+ */
+/datum/component/soulcatcher/proc/create_room(target_name = "default room", target_desc = "it's a room")
+	var/datum/soulcatcher_room/created_room = new(src)
+	created_room.name = target_name
+	created_room.room_description = target_desc
+	soulcatcher_rooms += created_room
+
+	created_room.master_soulcatcher = WEAKREF(src)
+
+///Recieves a message from a soulcatcher room.
+/datum/component/soulcatcher/proc/recieve_message(message_to_recieve)
+	if(!message_to_recieve)
+		return FALSE
+
+	if(istype(parent, /obj/item))
+		var/obj/item/parent_item = parent
+		var/mob/living/holder = parent_item.loc
+		if(!holder)
```

Comment:
Could possibly just be the `!istype()` but I can't remember off the top of my head how that behaves with nulls.
```suggestion
		var/mob/living/holder = parent_item.loc
		if(!holder || !istype(holder))
```

Reply:
tested it, istype(null) is FALSE